examples: Add example used in the ElasticON presentation (#122)
elastic/terraform-provider-ec: examples/deployment-with-init/README.md

# Deployment Example

> This example shows how to deploy an Elastic Cloud deployment using Terraform.
> It will also initialize the instance by using some of the outputs as string variables within a very simple bash script, to create a user and an index.

Reviewer: ```suggestion
This example shows how to create an Elastic Cloud deployment using Terraform.

Secondly, it will show you how to initialize the instance. This is done by using some of the outputs as string variables within a very simple bash script to create a user and an index.
```

Lastly, it also creates a traffic filter (which allows all traffic) and attaches that to the deployment as well.

## Running the example

> build the provider using `make install` from the main folder

Reviewer: ```suggestion
Build the provider using `make install` from the main folder
```
